students/mark/Session05/mailroom.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python3
+
+
+"""
+Session03 - Part I
+
+
+https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom.html#exercise-mailroom
+
+Write a small command-line script called mailroom.py. This script should be executable.
+The script should accomplish the following goals:
+
+    It should have a data structure that holds a list of your donors and a history of the
+    amounts they have donated. This structure should be populated at first with at least
+    five donors, with between 1 and 3 donations each.
+
+    You can store that data structure in the global namespace.
+
+    The script should prompt the user (you) to choose from a menu of 3 actions:
+    “Send a Thank You”, “Create a Report” or “quit”)
+
+
+
+### Finally, use only functions and the basic Python data types you’ve learned about so far.
+There is no need to go any farther than that for this assignment.
+
+/*
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+William Gates, III         $  653784.49           2  $   326892.24
+Mark Zuckerberg            $   16396.10           3  $     5465.37
+Jeff Bezos                 $     877.33           1  $      877.33
+Paul Allen                 $     708.42           3  $      236.14
+*/
+
+/*
+
+
+*/
+"""
+
+# donor_db = {"William Gates, III":[326888.82, 326895.67],
+#             "Mark Zuckerberg":[5565.37, 5465.37, 5365.36],
+#             "Jeff Bezos":[877.33],
+#             "paul allen":[236.14, 236.14, 236.14],
+#             "jose gonzalez" : [123.45, 678.90, 101.11]
+#             }
+
+donor_db = {"william gates, iii":[326888.82, 326895.67],
+            "mark zuckerberg":[5565.37, 5465.37, 5365.36],
+            "jeff bezos":[877.33],
+            "paul allen":[236.14, 236.14, 236.14],
+            "jose gonzalez" : [123.45, 678.90, 101.11]
+            }
+
+
+def format_donor_name(donor_names):
+    if ',' in donor_names:
+        name_title=donor_names.split(',')
+        i_var=name_title[0].title() + ','
+        donor_names=i_var.replace(',', "," + name_title[1].upper())
+        return(donor_names)
+    else:
+        name_title=donor_names
+        return(donor_names.title())
+
+
+def print_donor_name(donor_db):
+    """
+    Output the donor name(s)
+    :param: donor_db dict
+    :returns: zero for success
+    """
+    print("Donor list: \n")
+    for donor_names in donor_db.keys():
+        if ',' in donor_names:
+            name_title=donor_names.split(',')
+            i_var=name_title[0].title() + ','
+            donor_names=i_var.replace(',', "," + name_title[1].upper())
+            print(donor_names)
+        else:
+            name_title=donor_names
+            print(donor_names.title())
+    return 0
+
```

Comment:
Not necessary to return 0. If you aren't returning anything, the white space defines the end of the function so you can leave the return out. Python defaults to returning None if there is no return statement.